### PR TITLE
[EXP-84-284] fix: more convex fixes

### PR DESCRIPTION
**Pull request by jstashh**
* Ensure strategies are in the correct order - curve first, then convex
* Correct price calculation for `rKP3R`
* Handle some strategies using `id` instead of `pid`
* Handle some strategies using `localKeepCRV` rather than `KeepCRV`

### CODE REVIEW

File: yearn/apy/curve/simple.py [hunk 1]
```diff
@@ -299,13 +315,28 @@ def _get_reward_apr(self, cvx_strategy, cvx_booster, base_asset_price, pool_pric
...
+
+        for x in range(rewards_length):
+            virtual_rewards_pool = contract(rewards_contract.extraRewards(x))
+                # do this for all assets, which will duplicate much of the curve info but we don't want to miss anything
+            if virtual_rewards_pool.periodFinish() > current_time:
+                reward_token = virtual_rewards_pool.rewardToken()
+               
+                # if the reward token is rKP3R we need to calculate it's price in 
+                # terms of KP3R after the discount
+                if reward_token == addresses[chain.id]['rkp3r_rewards']:
```
Comment (jstashh) [inline yearn/apy/curve/simple.py:331]: forgot to add this custom logic for rKP3R

**Review by mariuspod — CHANGES_REQUESTED** [review]

File: yearn/apy/curve/simple.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jstashh) [reply yearn/apy/curve/simple.py]: Added a guard to the entry point of this file, as we access `chain.id` in multiple places - 42ba267

[the remark this answers is not in the document]
Comment (jstashh) [reply yearn/apy/curve/simple.py]: Refactored addresses[chain.id] into its own variable in 573fe42af66655ea11deee3a897d40312e9a4791

**Review by mariuspod — APPROVED** [review]
Tested manually and with some unit and integration tests here:
https://github.com/jstashh/yearn-exporter/pull/2

**Pull request merged**